lib/improver/tests/utilities/cube_manipulation_new/test_MergeCubes.py

```diff
@@ -0,0 +1,406 @@
+        cube3.cell_methods = tuple([self.cell_method1, self.cell_method3])
+        result = self.plugin._equalise_cell_methods(
+            iris.cube.CubeList([cube1, cube2, cube3]))
+        self.assertIsInstance(result, iris.cube.CubeList)
+        self.assertEqual(len(result[0].cell_methods), 1)
+        check = result[1].cell_methods[0] == self.cell_method1
+        self.assertTrue(check)
+
+
+class Test__check_time_bounds_ranges(IrisTest):
+    """Test the _check_time_bounds_ranges method"""
+
+    def setUp(self):
+        """Set up some cubes with different time bounds ranges"""
+        frt = dt(2017, 11, 9, 21, 0)
+        times = [dt(2017, 11, 10, 3, 0),
+                 dt(2017, 11, 10, 4, 0),
+                 dt(2017, 11, 10, 5, 0)]
+        time_bounds = np.array([
+            [dt(2017, 11, 10, 2, 0), dt(2017, 11, 10, 3, 0)],
+            [dt(2017, 11, 10, 3, 0), dt(2017, 11, 10, 4, 0)],
+            [dt(2017, 11, 10, 4, 0), dt(2017, 11, 10, 5, 0)]])
+
+        cubes = iris.cube.CubeList([])
+        for tpoint, tbounds in zip(times, time_bounds):
+            cube = set_up_probability_cube(
+                0.6*np.ones((2, 3, 3), dtype=np.float32),
+                np.array([278., 280.], dtype=np.float32),
+                time=tpoint, frt=frt, time_bounds=tbounds)
+            cubes.append(cube)
+        self.matched_cube = cubes.merge_cube()
+
+        time_bounds[2, 0] = dt(2017, 11, 10, 2, 0)
+        cubes = iris.cube.CubeList([])
+        for tpoint, tbounds in zip(times, time_bounds):
+            cube = set_up_probability_cube(
+                0.6*np.ones((2, 3, 3), dtype=np.float32),
+                np.array([278., 280.], dtype=np.float32),
+                time=tpoint, frt=frt, time_bounds=tbounds)
+            cubes.append(cube)
+        self.unmatched_cube = cubes.merge_cube()
+        self.plugin = MergeCubes()
+
+    def test_basic(self):
+        """Test no error when bounds match"""
+        self.plugin._check_time_bounds_ranges(self.matched_cube)
+
+    def test_inverted(self):
+        """Test no error when bounds ranges match but bounds are in the wrong
+        order"""
+        inverted_bounds = np.flip(
+            self.matched_cube.coord("time").bounds.copy(), axis=1)
+        self.matched_cube.coord("time").bounds = inverted_bounds
+        self.plugin._check_time_bounds_ranges(self.matched_cube)
+
+    def test_error(self):
+        """Test error when bounds do not match"""
+        msg = 'Cube with mismatching time bounds ranges'
+        with self.assertRaisesRegex(ValueError, msg):
+            self.plugin._check_time_bounds_ranges(self.unmatched_cube)
+
+    def test_no_error_missing_coord(self):
+        """Test missing time or forecast period coordinate does not raise
+        error"""
+        self.matched_cube.remove_coord("forecast_period")
+        self.plugin._check_time_bounds_ranges(self.matched_cube)
+
+
+class Test_process(IrisTest):
```

Comment:
I think you need unit tests for:
* a single cube in rather than a cubelist
* that your input cubes are unchanged
* That the self.coord_mismatch_error_keys are updated appropriately for a threshold coord

Reply:
OK the last of these took a little reworking, but I've now added tests to cover this.